### PR TITLE
feat(ingest/snowflake): generate query nodes for snowflake

### PR DESCRIPTION
**Pull request by mayurinehate**
Other noteworthy changes:
1. Removed view lineage fallback for views with failed sql parsing
2. Populated timestamp in Upstream class same as query time

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py [hunk 1]
```diff
@@ -305,8 +315,6 @@ def __init__(self, ctx: PipelineContext, config: SnowflakeV2Config):
         self.view_definitions: FileBackedDict[str] = FileBackedDict()
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py:315]: one of the neat things about the new sql aggregator is that it internally already has a FileBackedDict for view definitions, and defers their processing until all schemas have been loaded

so it's totally safe to just call .add_view_definition immediately, and never manually track the view definitions in the source itself

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py [hunk 2]
```diff
@@ -1689,7 +1676,8 @@ def close(self) -> None:
         super().close()
         StatefulIngestionSourceBase.close(self)
         self.view_definitions.close()
-        self.sql_parser_schema_resolver.close()
+        # Do we not need to close filebackeddicts opened explicitly ?
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py:1679]: it's nicer to close them explicitly, but they'll get automatically closed when the objects get garbage collected

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage_v2.py [hunk 3]
```diff
@@ -58,11 +60,47 @@
...
+    query_id: str
+
+
+class Query(TypedDict):
+    query_id: str
+    query_text: str
+    start_time: str
+
+
+class UpstreamLineageEdge(TypedDict):
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage_v2.py:91]: if we made these all pydantic objects, we could just call .parse_obj and not manually map everything

the JSON stuff could also get fixed up by a validator (similar to what we do here https://github.com/acryldata/acryl-executor/blob/5a3bbf92386a1af68661c5d0f55bd1bb437ea578/src/acryl/executor/execution/sub_process_task_common.py#L151)

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py [hunk 4]
```diff
@@ -236,9 +236,18 @@ def __init__(self, ctx: PipelineContext, config: SnowflakeV2Config):
 
         # For database, schema, tables, views, etc
         self.data_dictionary = SnowflakeDataDictionary()
-
         self.lineage_extractor: Optional[SnowflakeLineageExtractor] = None
+        self.aggregator: Optional[SqlParsingAggregator] = None
+
         if self.config.include_table_lineage:
+            self.aggregator = SqlParsingAggregator(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py:243]: I know `generate_usage_statistics=False` and `generate_operations=False` are the default, but it'd be good to set them explicitly for some extra code clarity

**Review by hsheth2 — COMMENTED** [review]
Mostly looks good, left a few minor suggestions

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage_v2.py (hunk 3 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage_v2.py]: Done

**Review by hsheth2 — APPROVED** [review]
LGTM

**Comment by hsheth2** [comment]
Lint issues should be fixed by https://github.com/datahub-project/datahub/pull/9976

**Pull request merged**